yarn-project/acir-simulator/src/client/private_execution.ts

```diff
@@ -197,6 +206,26 @@ export class PrivateFunctionExecution {
       storageRead: notAvailable,
       storageWrite: notAvailable,
       callPublicFunction: notAvailable,
+      emitEncryptedLog: ([acvmContractAddress, acvmStorageSlot, ownerX, ownerY, ...acvmPreimage]: ACVMField[]) => {
+        const contractAddress = AztecAddress.fromBuffer(fromACVMField(acvmContractAddress).toBuffer());
+        const storageSlot = fromACVMField(acvmStorageSlot);
+        const preimage = acvmPreimage.map(f => fromACVMField(f));
+
+        const notePreimage = new NotePreimage(preimage);
+        const txAuxData = new TxAuxData(notePreimage, contractAddress, storageSlot);
+
+        const owner = {
+          x: fromACVMField(ownerX),
+          y: fromACVMField(ownerY),
+        };
+        const ownerPublicKey = Point.fromBuffer(Buffer.concat([owner.x.toBuffer(), owner.y.toBuffer()]));
```

Comment:
seems like the x/y coordinates go through conversions like this: string -> Buffer -> Field -> Buffer.
Would be slightly more efficient to convert directly to a Buffer unless the conversion to Field is actually necessary?

Reply:
I agree. Will see if I can streamline it without readability loss.

Reply:
Addressed in [88ca1e3](https://github.com/AztecProtocol/aztec-packages/pull/777/commits/88ca1e3796ec2c3c29e44d16c637a8b8838c70e3)